android/app/src/main/java/io/github/parkwithease/parkeasy/ui/login/LoginViewModel.kt

```diff
@@ -29,7 +59,58 @@ class LoginViewModel @Inject constructor(private val repo: ParkEasyRepository) :
         _password.value = input
     }
 
+    fun onConfirmPasswordChange(input: String) {
+        _confirmPassword.value = input
+        _matchingPasswords.value = _password.value == _confirmPassword.value
+    }
+
     fun onLoginPress() {
-        runBlocking { login(Credentials(_email.value, _password.value)) }
+        runBlocking {
+            launch {
+                if (userRepo.login(LoginCredentials(_email.value, _password.value))) {
+                    _loggedIn.value = true
+                    _message.value = Event("Logged in successfully")
+                } else {
+                    _message.value = Event("Error logging in")
+                }
+            }
+        }
+    }
+
+    fun onRegisterPress() {
+        if (_password.value == _confirmPassword.value) {
+            runBlocking {
+                launch {
+                    if (
+                        userRepo.register(
+                            RegistrationCredentials(_name.value, _email.value, _password.value)
+                        )
+                    ) {
+                        _loggedIn.value = true
+                        _message.value = Event("Registered successfully")
+                    } else {
+                        _message.value = Event("Error registering")
+                    }
+                }
+            }
+        } else {
+            _message.value = Event("Passwords don't match")
+        }
+    }
+
+    fun onResetPress() {
+        runBlocking {
+            launch {
+                if (userRepo.reset(ResetCredentials(_email.value))) {
+                    _message.value = Event("Reset email sent\nJk... we're working on it")
+                } else {
+                    _message.value = Event("Error resetting password")
+                }
+            }
+        }
+    }
+
+    fun onSwitchPress() {
```

Comment:
You would probably want to reset all of your fields every time we switch state. This makes sure the form is clean for new action.

Reply:
I think that leaving the fields as they are is better since if the user tries to log in with an email and they don't have an account under that email, it's likely that they would use that same email to register a new account. Same goes with trying to log in and then trying to reset password